File: src/main/java/com/nurflugel/dependencyvisualizer/enums/OutputFormat.java
/**
 * Created by IntelliJ IDEA.
 * User: douglasbullard
 * Date: Dec 2, 2004
 * Time: 8:03:51 PM
 *
 */
package com.nurflugel.dependencyvisualizer.enums;

/** Copyright 2005, Nurflugel.com. */
public enum OutputFormat
{
  // ------------------------------ FIELDS ------------------------------
  Svg              ("SVG", "svg", ".svg"),
  Png              ("PNG", "png", ".png"),
  Pdf              ("PDF", "epdf", ".pdf"),
  PdfViaGhostscript("PDF_VIA_GHOSTSCRIPT", "ps", ".ps");

  private String displayLabel;  // for debug only
  private String extension;     // for debug only
  private String type;          // for debug only

  // --------------------------- CONSTRUCTORS ---------------------------
  /**  */
  OutputFormat(String displayLabel, String type, String extension)
  {
    this.displayLabel = displayLabel;
    this.type         = type;
    this.extension    = extension;
  }
  // -------------------------- OTHER METHODS --------------------------

  /**  */
  public String displayLabel()
  {
    return displayLabel;
  }

  /**  */
  public String extension()
  {
    return extension;
  }

  /**  */
  public String type()
  {
    return type;
  }
}

File: src/main/kotlin/com/nurflugel/dependencyvisualizer/enums/OutputFormat.kt
/**
 * Created by IntelliJ IDEA.
 * User: douglasbullard
 * Date: Dec 2, 2004
 * Time: 8:03:51 PM
 *
 */
package com.nurflugel.dependencyvisualizer.enums

/** Copyright 2005, Nurflugel.com.  */
enum class OutputFormat(
    // for debug only
     val displayLabel: String, // for debug only
     var type: String,
     var extension: String
) {
    // ------------------------------ FIELDS ------------------------------
    Svg("SVG", "svg", ".svg"),
    Png("PNG", "png", ".png"),
    Pdf("PDF", "pdf", ".pdf"),
    Dot("DOT", "dot", ".dot"),
    PdfViaGhostscript("PDF_VIA_GHOSTSCRIPT", "ps", ".ps");

    companion object {
        fun getDefaultExtension(): OutputFormat {
            return Pdf
        }
    }
}
